xknx/devices/sensor.py

```diff
@@ -79,6 +422,20 @@ def resolve_state(self) -> Any | None:
         """Return the current state of the sensor as a human readable string."""
         return self.sensor_value.value
 
+    @classmethod
+    def transcode_dpt_to_value_type(cls, dpt_id: str) -> int | str | None:
+        """Transcode DPT string to value type or return None if not supported."""
+        if not (dpt_cls := DPTBase.parse_transcoder(dpt_id)):
+            return None
+        return next(
+            (dpt.value_type for dpt in cls.supported_dpts if dpt == dpt_cls), None
+        )
```

Comment:
You can use `DPTNumeric.parse_transcoder() or DPTString.parse_transcoder()` to get HA SensorEntity compatible values.